### ISSUE TITLE
`%{gsub:...}` throws error

### ISSUE THREAD

**Issue by voxik** [rpm-software-management/rpm#3410]
From [documentation](https://github.com/rpm-software-management/rpm/blob/ebc406862053815528c00095f9f991d649fa33a0/docs/manual/macros.md?plain=1#L119), I would think this should work:

~~~
$ rpm --eval "%{gsub:foo o i}"
error: lua script failed: [string "gsub"]:1: bad argument #2 to 'gsub' (string expected, got no value)
~~~

Please note the example works just fine without the collon:

~~~
$ rpm --eval "%{gsub foo o i}"
fii
~~~

Testing on Fedora Rawhide:

~~~
$ rpm -q rpm
rpm-4.20.0-1.fc42.x86_64
~~~

**Comment by pmatilai** [comment]
Yeah both syntaxes should work. Thanks for reporting.

**Comment by mlschroe** [comment]
No, please do not change this. `%{macro:something}` is a way to expand a macro with exactly one argument. gsub needs more than one argument, thus the error.

**Comment by mlschroe** [comment]
See also:
```
$ rpm --eval '%{len foo bar}' --eval '%{len:foo bar}'
3
7
```

**Comment by pmatilai** [comment]
Hmm, I don't remember the : syntax being intentionally one argument only but then it's been a while.
But in that case this is a documentation issue.

**Comment by voxik** [comment]
> Hmm, I don't remember the : syntax being intentionally one argument only but then it's been a while. But in that case this is a documentation issue.

I'd argue this is just documentation issue. The "one arguments" syntax likely has some merit, but then the error message is quite unintuitive

**Comment by voxik** [comment]
> > Hmm, I don't remember the : syntax being intentionally one argument only but then it's been a while. But in that case this is a documentation issue.
> 
> I'd argue this is just documentation issue. The "one arguments" syntax likely has some merit, but then the error message is quite unintuitive

IOW, the error should say that the "one argument" syntax is not supported for `gsub`

**Comment by voxik** [comment]
> See also:
> 
> ```
> $ rpm --eval '%{len foo bar}'
> 3
> ```

And TBH, this is also quite questionable behavior.

**Comment by mlschroe** [comment]
That's caused by %len being mapped to lua's string.len()  function, and lua does not complain about the extra arg:
```
$ lua
> print(string.len("foo", "bar"))
3
> 
```
